chore: automate reproduction of examples

lee-chase opened the pull request:
The examples vary in the following ways:
1. The example usage (Which is all in the Example folder).
2. The prefix used
3. The use of flags (to enable a canary component).
4. The react version.

This PR uses a base `Templates` folder, using the same technique as the `generate` script to produce the common files for each example and set the variations noted as 2, 3 and 4 above.  In order to do so it reads the `gallery.config.json` file from each example folder, discarding everything other than the `thumbnail.png` and the contents of the `src/Example/*`.

A next step here is to upgrade the `scripts/example-gallery-builder/update-example/templates` to React 18.

NOTE: The update runs on each commit, as per the rest of the gallery update, we could move it to it's own github action to be run periodically.

elycheea commented on examples/carbon-for-ibm-products/PageHeader/package.json:
```diff
@@ -1,12 +1,14 @@
     "component",
-    "page-header"
+    "pageheader",
+		"page",
+		"header"
```
```suggestion
     "page",
     "header"
```

elycheea commented on examples/carbon-for-ibm-products/ProductiveCard/package.json:
```diff
@@ -1,12 +1,14 @@
     "component",
-    "productive-card"
+    "productivecard",
+		"productive",
+		"card"
```
```suggestion
     "productive",
     "card"
```

elycheea commented on examples/carbon-for-ibm-products/SidePanel/package.json:
```diff
@@ -1,12 +1,14 @@
     "component",
-    "sidepanel"
+    "sidepanel",
+		"side",
+		"panel"
```
```suggestion
     "side",
     "panel"
```

elycheea commented on examples/carbon-for-ibm-products/StatusIcon/package.json:
```diff
@@ -1,12 +1,14 @@
     "component",
-    "statusicon"
+    "statusicon",
+		"status",
+		"icon"
```
```suggestion
     "status",
     "icon"
```

elycheea commented on examples/carbon-for-ibm-products/TagSet/package.json:
```diff
@@ -1,12 +1,14 @@
     "component",
-    "tagset"
+    "tagset",
+		"tag",
+		"set"
```
```suggestion
     "tag",
     "set"
```

elycheea commented on examples/carbon-for-ibm-products/UserProfileImage/package.json:
```diff
@@ -1,12 +1,15 @@
     "component",
-    "userprofileimage"
+    "userprofileimage",
+		"user",
+		"profile",
+		"image"
```
```suggestion
     "user",
     "profile",
     "image"
```

elycheea commented on examples/carbon-for-ibm-products/prefix-example/package.json:
```diff
@@ -1,12 +1,14 @@
     "component",
-    "prefix"
+    "prefixexample",
+		"prefix",
+		"example"
```
```suggestion
     "prefix",
     "example"
```

elycheea commented on examples/carbon-for-ibm-products/react-16-example/package.json:
```diff
@@ -1,12 +1,15 @@
     "component",
-    "aboutmodal"
+    "react_16example",
+		"react",
+		"16",
+		"example"
```
```suggestion
     "react",
     "16",
     "example"
```

elycheea commented on examples/carbon-for-ibm-products/react-17-example/package.json:
```diff
@@ -1,12 +1,15 @@
     "component",
-    "aboutmodal"
+    "react_17example",
+		"react",
+		"17",
+		"example"
```
```suggestion
     "react",
     "17",
     "example"
```

elycheea commented on scripts/example-gallery-builder/update-example/index.js:
```diff
@@ -0,0 +1,153 @@
+    return;
+  }
+
+  //   fs.writeFileSync(
+  //     path,
+  //     `/**
+  //  * Copyright IBM Corp. 2023, 2023
+  //  *
+  //  * This source code is licensed under the Apache-2.0 license found in the
+  //  * LICENSE file in the root directory of this source tree.
+  //  */\n`
+  //   );
```
👀

Looks like we need to reorder some things here. Also seeing a `\n` in the copyright header.

lee-chase commented:
OK. @elycheea The tabs have been. replaced by spaces addressing the comments.

I have also:
- Removed pre-commit step
- Updated `pr.yml` github action check to check the gallery config is up to date (not sure how to test this without merging).
- Updated `update.yml` to also update the gallery config
- Updated `generate` script to also output a base example

elycheea reviewed (commented):
One typo. Noticed the examples are slightly downgraded, but I’m inclined to merge this and update the templates (e.g. next dependency update maybe??) instead of holding the changes in PR here.
